File: exemples/kata-bowling-csharp/KataBowling/Partie.cs
using System.Collections.Generic;

namespace KataBowling
{
    public class Partie
    {
        private readonly IList<int> _lancers = new List<int>();

        public void NouveauLancer(int quilles)
        {
            _lancers.Add(quilles);
        }

        public int CalculeScore()
        {
            var score = 0;
            var indiceLancer = 0;
            for (var i = 0; i < 10; i++)
            {
                if (EstStrike(indiceLancer))
                {
                    score += 10 + BonusStrike(indiceLancer);
                    indiceLancer++;
                }
                else if (EstSpare(indiceLancer))
                {
                    score += 10 + BonusSpare(indiceLancer);
                    indiceLancer += 2;
                }
                else
                {
                    score += _lancers[indiceLancer] + _lancers[indiceLancer + 1];
                    indiceLancer += 2;
                }
            }
            return score;
        }

        private bool EstStrike(int indiceLancer)
        {
            return _lancers[indiceLancer] == 10;
        }

        private int BonusStrike(int indiceLancer)
        {
            return _lancers[indiceLancer + 1] + _lancers[indiceLancer + 2];
        }

        private bool EstSpare(int indiceLancer)
        {
            return _lancers[indiceLancer] + _lancers[indiceLancer + 1] == 10;
        }

        private int BonusSpare(int indiceLancer)
        {
            return _lancers[indiceLancer + 2];
        }
    }
}

File: exemples/kata-bowling-java/src/main/java/Partie.java
import java.util.ArrayList;
import java.util.List;

public class Partie {
    public void nouveauLancer(int quilles) {
        lancers.add(quilles);
    }

    public int score() {
        int résultat = 0;
        int indiceCarreau = 0;
        for (int i = 0; i < 10; i++) {
            if (estStrike(indiceCarreau)) {
                résultat += 10 + bonusStrike(indiceCarreau);
                indiceCarreau++;
            } else if (estSpare(indiceCarreau)) {
                résultat += scoreCarreau(indiceCarreau) + bonusSpare(indiceCarreau);
                indiceCarreau += 2;
            } else {
                résultat += scoreCarreau(indiceCarreau);
                indiceCarreau += 2;
            }
        }
        return résultat;
    }

    private boolean estStrike(int i) {
        return lancers.get(i) == 10;
    }

    private int bonusStrike(int i) {
        return lancers.get(i + 1) + lancers.get(i + 2);
    }

    private boolean estSpare(int i) {
        return scoreCarreau(i) == 10;
    }

    private Integer bonusSpare(int i) {
        return lancers.get(i + 2);
    }

    private int scoreCarreau(int indiceCarreau) {
        return lancers.get(indiceCarreau) + lancers.get(indiceCarreau + 1);
    }

    private List<Integer> lancers = new ArrayList<Integer>();
}
